src/params/str-list.js

```diff
@@ -1,16 +1,28 @@
+const assert = require('assert');
 const List = require('./list');
 
 class StrList extends List {
-  constructor(...args) {
-    super(...args);
+  constructor(name, position, opts) {
+    super(name, position, opts);
     this.items = { type: 'string' };
+    if (opts !== undefined && opts.enums) {
+      assert(Array.isArray(opts.enums));
+      this.enums = new Set(opts.enums);
+    }
   }
 
   validate(value) {
     let valid = super.validate(value);
-    if (valid && (this.stringInput ? JSON.parse(value) : value).some(e => typeof e !== 'string')) {
-      valid = false;
+    if (valid) {
+      const strList = this.stringInput ? JSON.parse(value) : value;
```

Comment:
Have a look at `geo-shape.js`, and parse value similar to there.

The checks below are easier to understand if formatted similar to `geo-shape.js` as well, try not to nest if statements if not necessary